test/package_server.dart

```diff
@@ -220,13 +233,30 @@ class PackageServer {
         for (final e in contents ?? <d.Descriptor>[]) {
           addDescriptor(e, '');
         }
-        return Stream.fromIterable(entries)
+        return replaceOs(Stream.fromIterable(entries)
             .transform(tarWriterWith(format: OutputFormat.gnuLongName))
-            .transform(gzip.encoder);
+            .transform(gzip.encoder));
       },
     );
   }
 
+  /// Replaces the 9th entry in [stream] with a 0. This replaces the os entry
+  /// of a gzip stream, giving us the same stream on all platforms.
+  Stream<List<int>> replaceOs(Stream<List<int>> stream) async* {
```

Comment:
https://www.rfc-editor.org/rfc/rfc1952.html

And maybe explain a little... perhaps make this simpler by converting everyting to a single Uint8List and then modify it... and return a `Stream.single()`

Reply:
Maybe make it private.